onnxruntime/core/session/custom_ops.cc

```diff
@@ -83,6 +83,23 @@ struct CustomOpKernel : OpKernel {
   void* op_kernel_;
 };
 
+common::Status GetProviderString(OrtExecutionProviderType provider, std::string &providerString) {
+  static std::unordered_map<OrtExecutionProviderType, std::string> providerMap{
+    { ORT_EXECUTION_PROVIDER_CPU, kCpuExecutionProvider },
+    { ORT_EXECUTION_PROVIDER_CUDA, kCudaExecutionProvider },
+    { ORT_EXECUTION_PROVIDER_MKL_DNN, kMklDnnExecutionProvider },
+    { ORT_EXECUTION_PROVIDER_NUPHAR, kNupharExecutionProvider },
+    { ORT_EXECUTION_PROVIDER_BRAIN_SLICE, kBrainSliceExecutionProvider },
+    { ORT_EXECUTION_PROVIDER_TENSORRT, kTensorrtExecutionProvider },
+  };
+  if (providerMap.count(provider) > 0) {
```

Comment:
Thank you for making this as a function!

small comments: 1) will count and at do two map lookups? maybe using "find"? 2) do you want to avoid the string copy "providerString = providerMap.at(provider)"?

Reply:
@linkerzhang 1) You are right. I will change it to `find` (although I am not really a fan of iterators...). 2) I think this is by design because we don't want to pass pointer back. Defining `std::string *` outside and making the signature of the function as `common::Status(OrtExecutionProviderType, std::string **)` looks to be a bad idea because we shall avoid using pointers in C++. `unique_ptr`+`weak_ptr` will make the structure further more complex. What do you think the solution should be?

Reply:
IMHO, it would be best to use `std::string_view` in C++17. However that would break many old compilers...

Reply:
1) is now resolved by commit 5cca2a3.

Reply:
This discuss is no long valid as we changed from enum to string in the custom op interface for provider.